cdap-common/src/main/java/io/cdap/cdap/common/lang/jar/BundleJarUtil.java

```diff
@@ -191,6 +198,22 @@ public static File prepareClassLoaderFolder(Location jarLocation, File destinati
     return destinationFolder;
   }
 
+  private static void linkDirectories(File sourceDirectory, File targetDirectory) throws IOException {
+    File[] files = sourceDirectory.listFiles();
+    if (files == null) {
+      return;
+    }
+    for (File sourceFile : files) {
+      Path newTarget = targetDirectory.toPath().resolve(sourceFile.getName());
+      if (sourceFile.isDirectory()) {
+        newTarget.toFile().mkdirs();
```

Comment:
Use `Files.createDirectories(newTarget)` directly without converting to `File`.

Reply:
done